### PR TITLE
Issue/271 persist lang setting

### LINKED ISSUE #271: Persist Language
> We should persist the language selection.
Maybe we could add this to the current jwt token.

- [ ] Use `window.localStorage` to persits the current selected language

### PR DESCRIPTION
**Pull request by Paul-B98**
closes #271

### CODE REVIEW

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
In React-Land we should move this inside a Hook. What speaks against a simplified version of an [`useLocalStorage()`](https://usehooks-ts.com/react-hook/use-local-storage) or [`use-persisted-state`](https://github.com/donavon/use-persisted-state)? We can wrap `LocalStroageService` with such a hook.

**Comment by use-to** [comment]
@Paul-B98 I think @MaxSchaefer wants you to throw the `StorageSerice` away.

File: lib/hooks/useLocalStorage.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+declare global {
+  interface WindowEventMap {
+    'local-storage': CustomEvent
+  }
+}
+
+type SetValue<T> = Dispatch<SetStateAction<T>>
+type FunctionType = <T>(key: string, initValue: T) => [T, SetValue<T>]
+
+const useLocalStorage: FunctionType = <T, >(key: string, initValue: T) => {
```
Comment (MaxSchaefer) [inline lib/hooks/useLocalStorage.tsx:14]:
```suggestion
const useLocalStorage = <T, >(key: string, initValue: T): [T, SetValue<T>] => {
```

File: lib/hooks/useLocalStorage.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const get = useCallback((): T => {
+    if (typeof window === "undefined") {
```
Comment (MaxSchaefer) [inline lib/hooks/useLocalStorage.tsx:16]:
```suggestion
    if (typeof window === 'undefined') { 
```

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

**Comment by MaxSchaefer** [comment]
> @Paul-B98 I think @MaxSchaefer wants you to throw the `StorageSerice` away.

No, not necessarily. It's just an unusually OOP approach in our Frontend, the `hook`-Wrapper translates this very well into React-Land.

**Review by MaxSchaefer — APPROVED** [review]
@Paul-B98 looks good to me! Good job
I've just moved the call to `setStoredLanguage` directly into the `useLanguage`-Hook. Seams like a good idea that every call to `setLanguage` should persist the language as well.

**Comment by MaxSchaefer** [comment]
@Paul-B98 I've invited you to the web-Team. You can consider a squash merge.

**Pull request merged**